pkg/gator/read_constraints.go

```diff
@@ -154,3 +169,28 @@ func readConstraint(f fs.FS, path string) (*unstructured.Unstructured, error) {
 
 	return u, nil
 }
+
+// ReadK8sResources reads JSON or YAML k8s resources from an io.Reader,
+// decoding them into Unstructured objects and returning those objects as a
+// slice.
+func ReadK8sResources(r io.Reader) ([]*unstructured.Unstructured, error) {
+	var objs []*unstructured.Unstructured
+
+	decoder := yaml.NewYAMLOrJSONDecoder(r, 1000)
```

Comment:
nit: Use a named constant for `1000` with a name (or comment) that explains what the value is for. This way future editors of this file know if it's safe to modify, or what it may be reasonable to increase/decrease it to.